components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -1334,153 +1336,49 @@ void AtmosphereOutput::set_diagnostics()
 
 /* ---------------------------------------------------------- */
 std::shared_ptr<AtmosphereDiagnostic>
-AtmosphereOutput::create_diagnostic (const std::string& diag_field_name) {
-  auto& diag_factory = AtmosphereDiagnosticFactory::instance();
+AtmosphereOutput::create_diagnostic (const std::string& diag_field_name)
+{
+  // We need scream scope resolution, since this->create_diagnostic is hiding it
+  auto diag = scream::create_diagnostic(diag_field_name,get_field_manager("sim")->get_grid());
 
-  // Construct a diagnostic by this name
-  ekat::ParameterList params;
-  std::string diag_name;
+  // Some diags need some extra setup or trigger extra behaviors
```

Comment:
I think the gh diff may make it hard to see, with the red lines intertwined with the green lines. If you look at the file, it should be clear, since it's right there: we possibly set the mast value in the params list, and we trigger avg count.

Now, why we do those things, that may indeed not be obvious, so I'll add a comment.

Reply:
Yeah, FieldAtHeight and FieldAtPressure may request to find the value outside what's actually in the run (e.g., the latter for 1000mbar is OOB on mount Everest), which generates FillValue entries in the diag. When we do averaging, we don't want to count the FillValue, so we keep track of how many time samples are actuall _valid_ (and we keep a separate count for each column).

Reply:
Not all diags create fill value. If Aodvis does, then yes, we should handle that too. We could maybe think of adding a method to AtmosphereDiagnostic, something like "bool can_generate_fill_value()" so that we don't have to do all these if/else...